### PR TITLE
rainbowshuffling and bottom encoding

### PR DESCRIPTION
**Pull request by fly-san**
- added command `rainbowshuffle` which increases the speed of rainbow shuffles 👍 (see docstring for details)
- added commands `bottom` and `unbottom` which implement bottom encoding/decoding as specified in https://github.com/bottom-software-foundation/spec
- moved command `purge` to `admin.py`

i should probably start using branches :clueless: i just keep forgetting to checkout before i start on something new lol! 

GuMo

### CODE REVIEW

**Comment by fly-san** [comment]
meowww

**Comment by fly-san** [comment]
tl;dr: yes, but only if there's a catastrophic scale mass migration event where a lot of people join the server at the same time.

Explanation:

It is true that the spread may be unbalanced, but that is only perceptible if there's a large difference between the steps used on the previous shuffle and on the new shuffle.

The HSV.H value is currently calculated given `0.15 + i * (1 / n)`, where `i` is the member index and `n` is the total member count. 

In this case, `1/n` is the step between two colors, which means with a large enough `n`, the step will only significantly change for a large change in `n`.

An example, given our current member count of 73: the difference between `1/74` and `1/73` is just about 0.0185%.

Furthermore, the worst case scenario is if the server size doubles between two shuffles. In this case, half of the list will have a certain step while the other half will have a different step, but the rainbow effect will be maintained. (double rainbow, double gay)

In any case, the easiest solution is to simply use `$rainbowroles deactivate` first, followed by `$rainbowshuffle`.

**Comment by fly-san** [comment]
we could also just clamp `1/n` to a certain range i guess lol

**Comment by likeazir** [comment]
should be fine :)
